src/firmware_logger_device.h

```diff
@@ -1,66 +1,66 @@
 // License: Apache 2.0. See LICENSE file in root directory.
-// Copyright(c) 2020 Intel Corporation. All Rights Reserved.
+// Copyright(c) 2024 Intel Corporation. All Rights Reserved.
 
 #pragma once
 
 #include "core/extension.h"
 #include "device.h"
 #include "hw-monitor.h"
-#include <vector>
 #include "fw-logs/fw-log-data.h"
 #include "fw-logs/fw-logs-parser.h"
 
+#include <map>
+#include <string>
+
 namespace librealsense
 {
     class firmware_logger_extensions
     {
     public:
-        virtual bool get_fw_log(fw_logs::fw_logs_binary_data& binary_data) = 0;
-        virtual bool get_flash_log(fw_logs::fw_logs_binary_data& binary_data) = 0;
+        virtual bool get_fw_log( fw_logs::fw_logs_binary_data & binary_data ) = 0;
+        virtual bool get_flash_log( fw_logs::fw_logs_binary_data & binary_data ) = 0;
         virtual unsigned int get_number_of_fw_logs() const = 0;
-        virtual bool init_parser(std::string xml_content) = 0;
-        virtual bool parse_log(const fw_logs::fw_logs_binary_data* fw_log_msg, fw_logs::fw_log_data* parsed_msg) = 0;
+        virtual bool init_parser( std::string xml_content ) = 0;
+        virtual bool parse_log( const fw_logs::fw_logs_binary_data * fw_log_msg, fw_logs::fw_log_data * parsed_msg ) = 0;
         virtual ~firmware_logger_extensions() = default;
     };
-    MAP_EXTENSION(RS2_EXTENSION_FW_LOGGER, librealsense::firmware_logger_extensions);
+    MAP_EXTENSION( RS2_EXTENSION_FW_LOGGER, librealsense::firmware_logger_extensions );
 
     class firmware_logger_device : public virtual device, public firmware_logger_extensions
     {
     public:
         firmware_logger_device( std::shared_ptr< const device_info > const & dev_info,
-            std::shared_ptr<hw_monitor> hardware_monitor,
-            const command& fw_logs_command, const command& flash_logs_command);
+                                std::shared_ptr< hw_monitor > hardware_monitor,
+                                const command & fw_logs_command,
+                                const command & flash_logs_command,
+                                const std::map< int, std::string > & source_id_to_name = {} );
```

Comment:
What does it means
source id of what?

Is it possible we will have 0,DS5 as default or it is C API so we cannot?
trying to avoid the 20 {{0, "DS5"}}

Reply:
Source id is the name used in the struct definition. Source meaning the different SoCs of the camera.
Each camera model will declare the SoC specific to that model.

I considered using the legacy map as default but decided it was better not to do it in the API and to leave it empty.
I think I might have a way of getting the source map from the definition file, will try to do it and update.

Reply:
Getting the data from the definitions XML means we will have to use a different XML file for each camera model.
Lets talk about it tomorrow.

Reply:
As agreed, we will get the source map from the XML definitions, not hard coded per module.